packages/multi-column/src/test/commands/updateBlock.test.ts

```diff
@@ -0,0 +1,211 @@
+            },
+          ],
+        },
+      ],
+    });
+
+    expect(getEditor().document).toMatchSnapshot();
+  });
+
+  it("Update nested paragraph to column list", () => {
+    getEditor().updateBlock("nested-paragraph-0", {
+      type: "columnList",
+      children: [
+        {
+          type: "column",
+          children: [
+            {
+              type: "paragraph",
+              content: "Inserted Column Paragraph",
+            },
+          ],
+        },
+        {
+          type: "column",
+          children: [
+            {
+              type: "paragraph",
+              content: "Inserted Column Paragraph",
+            },
+          ],
+        },
+      ],
+    });
+
+    expect(getEditor().document).toMatchSnapshot();
+  });
+
+  it("Update column to column list", () => {
+    // should throw an error as we don't allow a column list inside a columnlist
+    expect(() => {
+      getEditor().updateBlock("column-0", {
+        type: "columnList",
+        children: [
+          {
+            type: "column",
+            children: [
+              {
+                type: "paragraph",
+                content: "Inserted Column Paragraph",
+              },
+            ],
+          },
+          {
+            type: "column",
+            children: [
+              {
+                type: "paragraph",
+                content: "Inserted Column Paragraph",
+              },
+            ],
+          },
+        ],
+      });
+    }).toThrow();
+  });
+
+  it("Update paragraph to column", () => {
+    getEditor().updateBlock("paragraph-0", {
+      type: "column",
+      children: [
+        {
+          type: "paragraph",
+          content: "Inserted Column Paragraph",
+        },
+      ],
+    });
+
+    expect(getEditor().document).toMatchSnapshot();
+  });
+
+  it("Update nested paragraph to column", () => {
+    getEditor().updateBlock("nested-paragraph-0", {
+      type: "column",
+      children: [
+        {
+          type: "paragraph",
+          content: "Inserted Column Paragraph",
+        },
+      ],
+    });
+
+    expect(getEditor().document).toMatchSnapshot();
+  });
+
+  it("Update column list to column", () => {
+    // this would cause a column to become a child of a node that's not a column list, and should thus throw an error
+    expect(() => {
+      getEditor().updateBlock("column-list-0", {
+        type: "column",
+        children: [
+          {
+            type: "paragraph",
+            content: "Inserted Column Paragraph",
+          },
+        ],
+      });
+    }).toThrow();
+  });
+
+  it("Update column list to paragraph", () => {
+    // this would cause columns to become children of a paragraph, and should thus throw an error
+    expect(() => {
+      getEditor().updateBlock("column-list-0", {
+        type: "paragraph",
+        content: "Inserted Column Paragraph",
+      });
+    }).toThrow();
+  });
+
+  it("Update column to paragraph", () => {
```

Comment:
I think this should throw an error same as updating a column list to a paragraph, since it violates the schema and doesn't really make sense

Reply:
Agree, will look into this